packages/nextjs/src/server/__tests__/clerkMiddleware.test.ts

```diff
@@ -113,6 +113,13 @@ describe('ClerkMiddleware type tests', () => {
     clerkMiddlewareMock();
   });
 
+  it('prevents usage of system permissions with auth.has()', () => {
+    clerkMiddlewareMock(async (auth, _event, _request) => {
+      // @ts-expect-error - system permissions are not allowed
+      (await auth()).has({ permission: 'org:sys_foo' });
+    });
+  });
+
```

Comment:
do you think we should add a test case for `useAuth()` where the type says that sys permissions are allowed ?

In order to not break something in the future